### PR TITLE
Fix taxon promotion rule form

### LINKED ISSUE #10813: Taxon promotion rule form problem
## Context

I can submit promotion rule of `Taxon`,  but after saved, it does not display that saved item.

See `Steps to Reproduce`.

This might be problem of `backend/app/views/spree/admin/promotions/rules/_taxon.html.erb`, I think.

## Expected Behavior

It displays that saved item(s).

expected

## Actual Behavior

After creating the rule, It does not display that saved item(s).

## Possible Fix

```diff
--- a/backend/app/views/spree/admin/promotions/rules/_taxon.html.erb
+++ b/backend/app/views/spree/admin/promotions/rules/_taxon.html.erb
@@ -1,7 +1,7 @@
 <div class="card-body">
   <div class="form-group taxons_rule_taxons">
-    <%= label_tag "#{param_prefix}_taxon_ids_string", Spree.t('taxon_rule.choose_taxons') %>
-    <%= select_tag "#{param_prefix}[taxon_ids_string]", nil, class: 'taxon_picker d-block w-100', id: :product_taxon_ids %>
+    <%= label_tag "#{param_prefix}_taxon_ids", Spree.t('taxon_rule.choose_taxons') %>
+    <%= select_tag "#{param_prefix}[taxon_ids]", options_from_collection_for_select(promotion_rule.taxons, :id, :pretty_name, promotion_rule.taxons.pluck(:id)), class: 'taxon_picker d-block w-100',multiple: true, id: :product_taxon_ids %>
   </div>
   <div class="form-group mb-0">
     <%= label_tag Spree.t('taxon_rule.label') %>
```

## Steps to Reproduce

1. Set taxon(s) in promotion rule of taxon form

2. Click ✔️Update

1st

3. Submitting form will succeed, but you cannot find saved taxon item.

2nd

## Your Environment

* Version used: spree4.2.0

### PR DESCRIPTION
**Pull request by hoshinotsuyoshi**
fixes https://github.com/spree/spree/issues/10813

### CODE REVIEW

File: backend/app/views/spree/admin/promotions/rules/_taxon.html.erb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class="card-body">
   <div class="form-group taxons_rule_taxons">
-    <%= label_tag "#{param_prefix}_taxon_ids_string", Spree.t('taxon_rule.choose_taxons') %>
-    <%= select_tag "#{param_prefix}[taxon_ids_string]", nil, class: 'taxon_picker d-block w-100', id: :product_taxon_ids %>
+    <%= label_tag "#{param_prefix}_taxon_ids", Spree.t('taxon_rule.choose_taxons') %>
+    <%= select_tag "#{param_prefix}[taxon_ids]", options_from_collection_for_select(promotion_rule.taxons, :id, :pretty_name, promotion_rule.taxons.pluck(:id)), class: 'taxon_picker d-block w-100', multiple: true, id: :product_taxon_ids %>
```
Comment (damianlegawiec) [inline backend/app/views/spree/admin/promotions/rules/_taxon.html.erb:4]:
```suggestion
    <%= select_tag "#{param_prefix}[taxon_ids]", options_from_collection_for_select(promotion_rule.taxons, :id, :pretty_name, promotion_rule.taxon_ids), class: 'taxon_picker d-block w-100', multiple: true, id: :product_taxon_ids %>
```

**Comment by damianlegawiec** [comment]
@hoshinotsuyoshi thank you for your contribution. I'm scratching my head on why our test suite didn't pick this up, could you add a regression test for this scenario?

File: backend/spec/features/admin/promotions/taxon_rule_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  let!(:taxon2) { taxons.second }
+
+  let!(:promotion) { create :promotion }
+
+  before do
+    visit spree.edit_admin_promotion_path(promotion)
+  end
+
+  context 'when there are no taxon rules' do
+    it 'adding a taxon rule', js: true do
```
Comment (hoshinotsuyoshi) [inline backend/spec/features/admin/promotions/taxon_rule_spec.rb:17]: regression test for `multiple: true`.

File: backend/spec/features/admin/promotions/taxon_rule_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      rule.update!(taxons: taxons)
+
+      visit spree.edit_admin_promotion_path(promotion)
+    end
+
+    it 'displays taxon names', js: true do
+      within('.promotion_rule') do
+        expect(page).to have_content(taxon1.name)
+        expect(page).to have_content(taxon2.name)
+      end
```
Comment (hoshinotsuyoshi) [inline backend/spec/features/admin/promotions/taxon_rule_spec.rb:47]: regression test for `select_tag`'s 2nd arg.

**Review by hoshinotsuyoshi — COMMENTED** [review]
👋  I added regression test with great reference to https://github.com/spree/spree/blob/v4.2.0/backend/spec/features/admin/promotions/option_value_rule_spec.rb .

@damianlegawiec

File: backend/app/views/spree/admin/promotions/rules/_taxon.html.erb (hunk 1 above)
Comment (hoshinotsuyoshi) [reply backend/app/views/spree/admin/promotions/rules/_taxon.html.erb]: thank you! applied https://github.com/spree/spree/pull/10814/commits/444cd03c8883ff0e8567edfcdbe7afee99020ebc

**Review by damianlegawiec — APPROVED** [review]
Awesome work @hoshinotsuyoshi 👍  We'll release this as part of 4.2.1 🚀

**Pull request merged**